src/test/java/seedu/address/logic/commands/UnmatchCommandTest.java

```diff
@@ -0,0 +1,196 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_JOB;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_PERSON;
+import static seedu.address.testutil.TypicalIndexes.INDEX_OUT_OF_BOUNDS;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_JOB;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_PERSON;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.Messages;
+import seedu.address.model.AddressBook;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.company.Company;
+import seedu.address.model.job.Job;
+import seedu.address.model.person.Person;
+import seedu.address.testutil.PersonBuilder;
+import seedu.address.testutil.TypicalCompanies;
+import seedu.address.testutil.TypicalJobs;
+import seedu.address.testutil.TypicalPersons;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for UnmatchCommand.
+ */
+public class UnmatchCommandTest {
+
+    private Model model;
+    private Model expectedModel;
+
+    @BeforeEach
+    public void setUp() {
+        model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+        expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+    }
+
+    @Test
+    public void constructor_nullIndex_throwsNullPointerException() {
+        Index validIndex = Index.fromOneBased(1);
+        assertThrows(NullPointerException.class, () -> new UnmatchCommand(null, null));
+        assertThrows(NullPointerException.class, () -> new UnmatchCommand(null, validIndex));
+        assertThrows(NullPointerException.class, () -> new UnmatchCommand(validIndex, null));
+    }
+
+    @Test
+    public void execute_validContactAndJob_unmatchSuccessful() throws Exception {
+        Index contactIndex = INDEX_FIRST_PERSON;
+        Index jobIndex = INDEX_FIRST_JOB;
+        // First, set up the model with an existing match
+        MatchCommand initialMatchCommand = new MatchCommand(contactIndex, jobIndex);
```

Comment:
This is supposed to be a unit test for UnmatchCommand, do not use actual MatchCommand here.

Reply:
I agree, you can probably try to use a model stub to manually match the contact and job together instead.

Reply:
@Green-Tea-123 allows me to chime in, using an actual model is allow in integration test, which @AlekKwek18 has specified in the JavaDoc of this test file. 

My concern was that no integration between MatchCommand and UnmatchCommand is needed, therefore it should not be used here.  

As far as `UnmatchCommand` is concerned, it will operate on model with match association without knowing the existence of `MatchCommand`, therefore I think use the helper function `setModelAfterMatching()` to setup the actual model for testing is fine. 

If @AlekKwek18 prefers a pure unit test it works too, you may find `testutil/ModelStub` helpful.